projects/ngx-d3timeline/src/lib/ngx-d3timeline.service.ts

```diff
@@ -61,6 +64,31 @@ export class NgxD3TimelineService {
     }
   }
 
+  componentDestroyed() {
+    this.deystroySubject.next(true);
+  }
+
+  onHovered(hovered: EventEmitter<Activity>) {
+    this.outputOnObservableEmit(this.hoveredActivity$, hovered);
+  }
+
+  onUnhovered(unhovered: EventEmitter<Activity>) {
+    this.outputOnObservableEmit(this.unhoveredActivity$, unhovered);
+  }
+
+  onActivityDropped(activityDropped: EventEmitter<Activity>) {
+    this.outputOnObservableEmit(this.activityDropped$, activityDropped);
+  }
+
+  private outputOnObservableEmit<T>(
```

Comment:
I reckon this should go into core, even though its only used here it is generic in nature, there is nothing in it specific to this service/component or even the timeline in general.